sim/priest/priest.go

```diff
@@ -206,3 +257,190 @@ func New(char *core.Character, selfBuffs SelfBuffs, talents string) *Priest {
 type PriestAgent interface {
 	GetPriest() *Priest
 }
+
+// Priest Specific Aura MOD Handling
+// Other flags
+type PriestSpellFlag uint64
+
+// Returns whether there is any overlap between the given masks.
+func (se PriestSpellFlag) Matches(other PriestSpellFlag) bool {
+	return (se & other) != 0
+}
+
+const (
+	PriestSpellFlagNone  PriestSpellFlag = 0
+	PriestSpellArchangel PriestSpellFlag = 1 << iota
+	PriestSpellDarkArchangel
+	PriestSpellBindingHeal
+	PriestSpellCircleOfHealing
+	PriestSpellDevouringPlague
+	PriestSpellDesperatePrayer
+	PriestSpellDispersion
+	PriestSpellDivineHymn
+	PriestSpellFade
+	PriestSpellFlashHeal
+	PriestSpellGreaterHeal
+	PriestSpellGuardianSpirit
+	PriestSpellHolyFire
+	PriestSpellHolyNova
+	PriestSpellHolyWordChastise
+	PriestSpellHolyWordSanctuary
+	PriestSpellHolyWordSerenity
+	PriestSpellHymnOfHope
+	PriestSpellImprovedDevouringPlague
+	PriestSpellInnerFire
+	PriestSpellInnerFocus
+	PriestSpellInnerWill
+	PriestSpellManaBurn
+	PriestSpellMindBlast
+	PriestSpellMindFlay
+	PriestSpellMindSear
+	PriestSpellMindSpike
+	PriestSpellMindTrauma
+	PriestSpellPainSuppresion
+	PriestSpellPenance
+	PriestSpellPowerInfusion
+	PriestSpellPowerWordBarrier
+	PriestSpellPowerWordShield
+	PriestSpellPrayerOfHealing
+	PriestSpellPrayerOfMending
+	PriestSpellPsychicScream
+	PriestSpellRenew
+	PriestSpellShadowOrbPassive
+	PriestSpellShadowWordDeath
+	PriestSpellShadowWordPain
+	PriestSpellShadowFiend
+	PriestSpellShadowyApparation
+	PriestSpellSmite
+	PriestSpellVampiricEmbrace
+	PriestSpellVampiricTouch
+
+	PriestSpellDoT     = PriestSpellDevouringPlague | PriestSpellHolyFire | PriestSpellMindFlay | PriestSpellShadowWordPain | PriestSpellVampiricTouch
+	PriestSpellInstant = PriestSpellCircleOfHealing |
+		PriestSpellDesperatePrayer |
+		PriestSpellDevouringPlague |
+		PriestSpellFade |
+		PriestSpellGuardianSpirit |
+		PriestSpellHolyNova |
+		PriestSpellHolyWordChastise |
+		PriestSpellHolyWordSanctuary |
+		PriestSpellHolyWordSerenity |
+		PriestSpellInnerFire |
+		PriestSpellPainSuppresion |
+		PriestSpellPowerInfusion |
+		PriestSpellPowerWordBarrier |
+		PriestSpellPowerWordShield |
+		PriestSpellRenew |
+		PriestSpellShadowWordDeath |
+		PriestSpellShadowWordPain |
+		PriestSpellVampiricEmbrace
+)
+
+type PriestAuraMod[T any] struct {
+	ClassSpell PriestSpellFlag
+	School     core.SpellSchool
+	BaseValue  T
+
+	// dynamic evaluation will be added to the given BaseValue
+	DynamicValue func(*Priest) T
+
+	// Stacks * (BaseValue + DynamicValue)
+	Stacks  int32
+	SpellID int32
+}
```

Comment:
The concept of tracking these aura mods and the various helpers makes sense to me but I wonder if this should be something we make general-purpose for all classes rather than just shadow-specific?

Reply:
Yeah it's how at least p-servers handle those mods internally.

https://wago.tools/db2/SpellClassOptions?build=4.4.0.54027&filter[SpellID]=exact%3A2944&page=1 - Class mask for Devouring 
https://wago.tools/db2/SpellEffect?build=4.4.0.54027&filter[SpellID]=63626&page=1 - Improved Devouring Plague Talent has the matching class mask - thus the dummy (usually the backing lua script) is only applied to that ability.

For now I've 'hacked' the damage mod applications by putting a wrapper around the spell config and changing the respective fields of the spell before calling the tick / apply methods. So it's not realy working nicely with the PseudoStats currently 😁

Reply:
I don't like this implementation at all. It completely ignores what the core supports and implements the same thing but specific to the priest. It also makes unneccessery calculations during ApplyEffects which can be done once during initialization using the spell configs damage multipliers. 
I can see why this is how you want to do it on an actual server where those things need to be calculated dynamically but there is no reason to do it like this in the sim. We want all static modifiers to be calculated once on sim creation and not done on every application when you run thousends of iterations.

Reply:
Totally get that. Was easier for me to get a handle of things as keeping track of everything for every spell was a bit overwhelming at first, but after I have implemented everything for shadow now I can refactor everything into the respective spells / PseudoStats. Will push a commit later for that!

Reply:
We can still have helper methods for building the damage mods in a way similiar to what you have done as long as its done outside of the ApplyEffects. Also as @kayla-glick was saying we may want to consider making this a part of core if people favour building the mods in such a way (Instead of what we currently do with having it explicitly build in their registration methods as I also agree with your opinion of this being overwhelming when you do the initial implementation)